uk.co.compucorp.civicrm.hrcomments/CRM/HRComments/BAO/Comment.php

```diff
@@ -8,16 +10,81 @@ class CRM_HRComments_BAO_Comment extends CRM_HRComments_DAO_Comment {
    * @param array $params key-value pairs
    * @return CRM_HRComments_DAO_Comment|NULL
    */
-  public static function create($params) {
+  public static function create($params, $validate = true) {
     $entityName = 'Comment';
     $hook = empty($params['id']) ? 'create' : 'edit';
 
     CRM_Utils_Hook::pre($hook, $entityName, CRM_Utils_Array::value('id', $params), $params);
+
+    if($validate){
+      self::validateParams($params);
+    }
+
     $instance = new self();
     $instance->copyValues($params);
+
+    if($hook == 'create') {
+      $instance->created_at = CRM_Utils_Date::processDate('now');
+    }
+
```

Comment:
I just realized that, the way it is now, it will be possible to change the `create_at` with an update:

```php
civicrm_api3('Comment', 'create', [
  'id' => 1,
  'create_at' => 'some-new-date'
]);
```

I don't think we should allow this. Probably the simplest way to accomplish that is by always removing `created_at` from `$params` (it's fine to remove it on create, because we'll automatically set it to the current date)